## 38_hurd.patch by  <glandium@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix FTBFS on Hurd because of MAXPATHLEN

Index: nss/mozilla/security/nss/cmd/shlibsign/shlibsign.c
===================================================================
--- nss.orig/mozilla/security/nss/cmd/shlibsign/shlibsign.c	2013-02-15 09:16:31.533595971 +0100
+++ nss/mozilla/security/nss/cmd/shlibsign/shlibsign.c	2013-02-15 09:16:31.525596162 +0100
@@ -707,7 +707,6 @@
 #ifdef USES_LINKS
     int ret;
     struct stat stat_buf;
-    char link_buf[MAXPATHLEN+1];
     char *link_file = NULL;
 #endif
 
@@ -1035,10 +1034,22 @@
     }
     if (S_ISLNK(stat_buf.st_mode)) {
         char *dirpath,*dirend;
-        ret = readlink(input_file, link_buf, sizeof(link_buf) - 1);
-        if (ret < 0) {
-            perror(input_file);
-            goto cleanup;
+        char *link_buf = NULL;
+        size_t size = 64;
+        while (1) {
+            link_buf = realloc(link_buf, size);
+            if (!link_buf) {
+                perror(input_file);
+                goto cleanup;
+            }
+            ret = readlink(input_file, link_buf, size - 1);
+            if (ret < 0) {
+                perror(input_file);
+                goto cleanup;
+            }
+            if (ret < size - 1)
+                break;
+            size *= 2;
         }
         link_buf[ret] = 0;
         link_file = mkoutput(input_file);
Index: nss/mozilla/security/nss/lib/freebl/unix_rand.c
===================================================================
--- nss.orig/mozilla/security/nss/lib/freebl/unix_rand.c	2013-02-15 09:16:31.533595971 +0100
+++ nss/mozilla/security/nss/lib/freebl/unix_rand.c	2013-02-15 09:16:31.525596162 +0100
@@ -1020,6 +1020,10 @@
 #define _POSIX_PTHREAD_SEMANTICS
 #include <dirent.h>
 
+#ifndef PATH_MAX
+#define PATH_MAX 1024
+#endif
+
 PRBool
 ReadFileOK(char *dir, char *file)
 {
